### PR TITLE
Feat/Node: Validation Block

### LINKED ISSUE #872: Add ValidationBlock - nodejs
https://github.com/iotaledger/tips-draft/blob/tip46/tips/TIP-0046/tip-0046.md#validation-block

### PR DESCRIPTION
**Pull request by kwek20**
Fixes #872

Blocked by #1001

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Can solve conflicts and merge now

File: bindings/nodejs/lib/types/block/core/block.ts [hunk 1]
```diff
@@ -40,4 +43,25 @@ export abstract class Block {
...
+        return this.type === BlockType.Validation;
+    }
+
+    /**
+     * Gets the block as an actual `ValidationBlock`.
+     * NOTE: Will throw an error if the block is not a `ValidationBlock`.
+     * @returns The block
+     */
+    asValidation(): ValidationBlock {
+        if (this.isBasic()) {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/core/block.ts]: 👀

File: bindings/nodejs/lib/types/block/core/validation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { HexEncodedString } from '../../utils';
+import { Block } from './block';
+
+/**
+ * A Validation Block is a special type of block used by validators to secure the network.
+ * It is recognised by the Congestion Control of the IOTA 2.0 protocol and can be issued without
+ * burning Mana within the constraints of the allowed validator throughput.
+ *
+ * It is allowed to reference more parent blocks than a normal Basic Block.
+ * Validation Block uses the introduced Block ID as its identifier.
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/core/validation.ts]: I think this sentence makes sense in the TIP but not here, I'd remove.

File: bindings/nodejs/lib/types/block/core/block.ts (hunk 1 above)
Comment (kwek20) [reply bindings/nodejs/lib/types/block/core/block.ts]: also missed it at wrapper. solved

File: bindings/nodejs/lib/types/block/core/validation.ts (hunk 2 above)
Comment (kwek20) [reply bindings/nodejs/lib/types/block/core/validation.ts]: which part specifically? the second part with It is.... its identifier?

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/core/validation.ts]: Yes, second sentence, we removed it in Rust already

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/block/core/validation.ts]: Sorry I mean second sentence of the second part! The last sentence.

Comment (kwek20) [reply bindings/nodejs/lib/types/block/core/validation.ts]: ill copy from rust 😆

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**